### PR TITLE
Stabilize match_default_bindings

### PR DESCRIPTION
**Pull request by cramertj**
This includes a submodule update to rustfmt
in order to allow a stable feature declaration.

r? @nikomatsakis 

cc https://github.com/rust-lang/rust/issues/42640

Many of the tests this PR touches are merely testing the current lack of desired future behavior around https://github.com/rust-lang/rust/issues/44849 and https://github.com/rust-lang/rust/issues/44848 (cc @tschottdorf). I noticed the bullets for those items were checked on the tracking issue-- I've unchecked them, as they don't appear to have been completed and I don't see any comments indicating that we don't want to pursue them further. Still, I think it's fine to stabilize the current behavior, as I think expanding it in the future should be backwards-compatible.

### CODE REVIEW

**Comment by petrochenkov** [comment]
The tracking issue has two unresolved questions (https://github.com/rust-lang/rust/issues/44848, https://github.com/rust-lang/rust/issues/44849) - how they are currently treated? Is there some conservative future-compatible behavior implemented?

**Comment by cramertj** [comment]
@petrochenkov See my comment above-- these are both errors now, and I believe they could be made to *not* error in the future in a backwards-compatible way.

File: src/libsyntax/feature_gate.rs [hunk 1]
```diff
@@ -563,6 +560,8 @@ declare_features! (
     (accepted, conservative_impl_trait, "1.26.0", Some(34511), None),
     // The `i128` type
     (accepted, i128_type, "1.26.0", Some(35118), None),
+    // Default match binding modes (RFC 2005)
+    (accepted, match_default_bindings, "1.22.0", Some(42640), None),
```
Comment (kennytm) [inline src/libsyntax/feature_gate.rs]: This should definitely not be 1.22.0...

**Comment by kennytm** [comment]
@bors retry 

3 hour timeout

**Comment by kennytm** [comment]
@cramertj I think you need to actually rebase.

**Comment by cramertj** [comment]
@kennytm Weird-- I didn't see any conflicts. Rebased anyways.

@bors r=nikomatsakis p=15

**Comment by Mark-Simulacrum** [comment]
I will assume that this is spurious, but I suppose we should also cc @BurntSushi.

@bors retry

```
[01:54:53] thread 'test_cat::prop_cat_cols' panicked at '[quickcheck] TEST FAILED (runtime error). Arguments: (CsvData { data: [[[]]] }, CsvData { data: [[[239, 187, 191]]] })
[01:54:53] Error: "assertion failed: `(left == right)`\n  left: `[]`,\n right: `[[\"\", \"\\u{feff}\"]]`"', /cargo/registry/src/github.com-1ecc6299db9ec823/quickcheck-0.4.1/src/tester.rs:147:28
[01:54:53]
[01:54:53]
[01:54:53] failures:
[01:54:53]     test_cat::prop_cat_cols
[01:54:53]
[01:54:53] test result: FAILED. 421 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
[01:54:53]
[01:54:53]
[01:54:53] error: test failed, to rerun pass '--test tests'
[01:54:53] thread 'main' panicked at 'tests failed for https://github.com/BurntSushi/xsv', tools/cargotest/main.rs:100:9
```

**Comment by kennytm** [comment]
The xsv error is supposed to be fixed by BurntSushi/xsv@92de2886330b831d48a7072fbcd0ec55d000207d and #45550, not sure why this is coming back...

**Comment by BurntSushi** [comment]
@Mark-Simulacrum I would say that is definitely spurious. I'll try to take a closer look soonish, but not sure exactly when.

**Pull request merged**